pixi.toml

```diff
@@ -0,0 +1,160 @@
+# ########### #
+# BOILERPLATE #
+# ########### #
+
+[project]
+name = "obelisk"
+authors = [
+    "Albert Li <alberthli@caltech.edu>",
+    "Zach Olkin <zolkin@caltech.edu>",
+]
+description = "The AMBER Lab's generic robot control interface."
+channels = ["conda-forge"]
+platforms = ["linux-64"]
+
+[activation]
+scripts = ["obelisk_ws/install/setup.sh"]
+
+
+# ############ #
+# FEATURE SETS #
+# ############ #
+
+# minimal
+[dependencies]
+python = ">=3.10"
+pip = ">=24.0"
+
+[pypi-dependencies]
+matplotlib = ">=3.9.0"
+numpy = ">=1.26"
+obelisk_py = { path="./obelisk/python", editable=true }
+
+
+# TODO(ahl): once obelisk is stable, check if cuda deps are needed
+# cuda
+[feature.cuda118]
+channels = ["nvidia", "nvidia/label/cuda-11.8.0"]
+system-requirements = {cuda="11.8.0"}
+
+[feature.cuda118.dependencies]
+cuda = { version="*", channel="nvidia/label/cuda-11.8.0" }
+
+[feature.cuda121]
+channels = ["nvidia", "nvidia/label/cuda-12.1.0"]
+system-requirements = {cuda="12.1.0"}
+
+[feature.cuda121.dependencies]
+cuda = { version="*", channel="nvidia/label/cuda-12.1.0" }
+
+
+# torch
+[feature.torch118]
+channels = ["nvidia", "pytorch"]
+dependencies = { pytorch-cuda="11.8.0" }
+pypi-dependencies = { torchvision=">=0.18.1", tqdm=">=4.66.4" }
+
+[feature.torch121]
+channels = ["nvidia", "pytorch"]
+dependencies = {pytorch-cuda="12.1.0"}
+pypi-dependencies = { torchvision=">=0.18.1", tqdm=">=4.66.4" }
+
+
+# ros2
+[feature.ros2base]
+channels = ["robostack-staging"]
+
+[feature.ros2base.dependencies]
+ros-humble-ros-base = "*"
+colcon-common-extensions = "*"
+
+# TODO(ahl): this was just a test to see if we could install many
+# prune these away to a true minimal set later
+# [NOTE] on humble, we should expect python 3.10
+ros-humble-control-msgs = "*"
+ros-humble-xacro = "*"
+ros-humble-angles = "*"
+ros-humble-ros2-control = "*"
+ros-humble-realtime-tools = "*"
+ros-humble-control-toolbox = "*"
+ros-humble-moveit = "*"
```

Comment:
should we remove some of these extra dependencies (like moveit) now? We can always do that at a later PR.

Reply:
as per f2f, left commented out for now